circuit/js/src/scaffold.ts

```diff
@@ -173,6 +179,36 @@ export abstract class AxiomBaseCircuitScaffold<T> extends BaseCircuitScaffold {
     return computeQuery;
   }
 
+  buildMockComputeQuery() {
+    // Mock compute query only works for the following DEFAULT_CIRCUIT_CONFIG:
+    if (
+      DEFAULT_CIRCUIT_CONFIG.k !== 14 ||
+      DEFAULT_CIRCUIT_CONFIG.numAdvice !== 4 ||
+      DEFAULT_CIRCUIT_CONFIG.numLookupAdvice !== 1 ||
+      DEFAULT_CIRCUIT_CONFIG.numInstance !== 1 ||
+      DEFAULT_CIRCUIT_CONFIG.numLookupBits !== 13 ||
+      DEFAULT_CIRCUIT_CONFIG.numVirtualInstance !== 2
+    ) {
+      throw new Error(`MockComputeQuery not valid for this DEFAULT_CIRCUIT_CONFIG`);
+    }
```

Comment:
@yi-sun @jonathanpwang is there a calculation i could use to get the size of the vk or the length of the computeproof instead of relying on this default setup?

Reply:
`k` and `numLookupBits` may change depending on the circuit (they are auto-configured), so don't check those in this if statement. `numVirtualInstance` is also an internal number, but it doesn't hurt to check it (the real num instance is always 1 I believe).

I did finally think about the formula. Assuming the `numAdvice` and `numLookupAdvice` are not both 1 (in which case there is a special optimization that messes up this formula), it is:
* `num_fixed_columns = num_advice + num_fixed + 2 (range check lookup table)`
* `permutation columns = num_advice + num_lookup_advice + num_fixed`
* `preprocessed_len = num_fixed_columns + permutation_columns`
* `vk_len = 1 (metadata) + 1 (transcript initial state) + preprocessed_len`

In practice we always have `num_fixed = 1`, so the final formula is:
* `vk_len = 2 + 2*num_advice + num_lookup_advice + 2 * num_fixed + 2 = 6 + 2 * num_advice + num_lookup_advice = 15`, which matches your current constant